### PR TITLE
Fix duplicate entries Missing and not Missing

### PR DESCRIPTION
**Pull request by BobSilent**
I have the issue as well as described here: 

fix: #87
fix: [[Issue]: Missing Episode Fetcher Seems to Create Double Season Folders #9999](https://github.com/jellyfin/jellyfin/issues/9999)
or here https://forum.jellyfin.org/t-jellyfin-displaying-duplicate-missing-files-for-episodes
https://www.reddit.com/r/jellyfin/comments/t4z9jf/show_missing_episode_showing_episodes_i_already/

improves: Unhelpful error messages #71 

mainly if the items were added as missing first, they never get deleted after adding media files.

### CODE REVIEW

**Comment by crobibero** [comment]
@BobSilent we just merged in tvdb v4 changes, if this PR is still needed please update

**Comment by BobSilent** [comment]
@crobibero I need to test and check, do I have some time (I think latest until the weekend). 
Hopefully it can make it into the release then.

**Comment by BobSilent** [comment]
@crobibero I am currently facing some issues with #93, I will prepare a separate PR for these bug fixes!

**Comment by BobSilent** [comment]
please see #110 

will rebase this PR on #110

File: Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs [hunk 1]
```diff
@@ -113,6 +122,7 @@ private bool IsEnabledForLibrary(BaseItem item)
 
             if (series == null)
             {
+                _logger.LogDebug("Given input is no " + nameof(Series) + ", " + nameof(Season) + " or " + nameof(Episode) + ": {Type}", item.GetType());
```
Comment (crobibero) [inline Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs:125]: Use log templating
https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template

**Review by crobibero — CHANGES_REQUESTED** [review]

**Comment by BobSilent** [comment]
@crobibero  a little bit off topic for this PR, but I have prepared another cleanup and I saw this comment here
https://github.com/jellyfin/jellyfin-plugin-tvdb/blob/0fae3719c2fe75182bfe4faa33ba4c0c155eaa36/Jellyfin.Plugin.Tvdb/TvdbClientManager.cs#L395-L401

Are the mentioned "issues upstream" the fact that "suddenly" logging is no longer working as configured?
This cause i could identify:
the service collection is configured from the plugins quite early: 
https://github.com/jellyfin/jellyfin/blob/e93d03d8cbff2122d7296f477604146f64758a73/Jellyfin.Server/Program.cs#L188-L189

But Serilog is added quite late.
https://github.com/jellyfin/jellyfin/blob/e93d03d8cbff2122d7296f477604146f64758a73/Jellyfin.Server/Program.cs#L349-L351

if this is changed like that:
image
the `AddHttpClient` can be used, as internally the `AddHttpClient` calls `AddLogging()` and this will be called before adding serilog, therefore the configuration gets lost and only parts will be logged afterwards.
Shall I create a PR for this in `jellyfin/jellyfin` for the `10.8` branch?

**Comment by crobibero** [comment]
> Shall I create a PR for this in jellyfin/jellyfin for the 10.8 branch?

Thank you for investigating the figuring out the logging issue! We aren't planning on making any more 10.8 releases, so a PR targeting master would be greatly appreciated!

> Post it as separate issue here: crobibero/tvdb-sdk-csharp?

That sdk is 99.97% auto-generated, any schema issues should be filed in the upstream repository (https://github.com/thetvdb/v4-api). The difficulty there lies with trying to figure out which model the issue is actually in

I think this PR can be merged without fixing the deserialization issues

**Comment by BobSilent** [comment]
> I think this PR can be merged without fixing the deserialization issues

Yes, this is more like a fyi...

> Thank you for investigating the figuring out the logging issue! We aren't planning on making any more 10.8 releases, so a PR targeting master would be greatly appreciated!

Ok, first need to setup a system here on my side, as I saw there are also other (bigger) changes in, like https://github.com/jellyfin/jellyfin/pull/9078

**Comment by BobSilent** [comment]
Shall I rebase to latest master?

**Comment by crobibero** [comment]
I'm not sure what you're asking but I'm fairly certain that the issue still persists in latest master of jellyfin/jellyfin

**Comment by BobSilent** [comment]
sorry maybe confusing due too many topics here ;)

I just wanted to know if i shall rebase this PR to latest?

**Comment by crobibero** [comment]
Don't worry about rebasing this PR, I'm going to start reviewing it shortly

File: Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs [hunk 2]
```diff
@@ -125,16 +135,20 @@ private void OnProviderManagerRefreshComplete(object? sender, GenericEventArgs<B
         {
             if (!IsEnabledForLibrary(genericEventArgs.Argument))
             {
+                _logger.LogDebug("{ProviderName} not enabled for {InputName}", ProviderName, genericEventArgs.Argument.Name);
                 return;
             }
 
+            _logger.LogDebug(nameof(OnProviderManagerRefreshComplete) + ": Try Refreshing for Item {Name} {Type}", genericEventArgs.Argument.Name, genericEventArgs.Argument.GetType());
```
Comment (crobibero) [inline Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs:142]: These should also be moved to templated logging

**Review by crobibero — COMMENTED** [review]
still have some templated logging fixes to be made

**Comment by scampower3** [comment]
> @crobibero during testing I saw some issues with the `Tvdb.Sdk` dll. Sometimes I get `System.Text.Json.JsonException: The JSON value could not be converted...` Exceptions as the schema seems to be quite strict, where it seems that TVDB send `null` instead of a value:
> 
> ```
> Tvdb.Sdk.SearchException: Could not deserialize the response body stream as Tvdb.Sdk.Response45.
> 
> Status: 200
> Response: 
> 
>  ---> System.Text.Json.JsonException: The JSON value could not be converted to System.Int32. Path: $.data[0].movie.runtime | LineNumber: 0 | BytePositionInLine: 281.
>  ---> System.InvalidOperationException: Cannot get the value of a token type 'Null' as a number.
>    at System.Text.Json.Utf8JsonReader.TryGetInt32(Int32& value)
>    at System.Text.Json.Utf8JsonReader.GetInt32()
>    at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
>    at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
>    at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
>    at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
>    at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
>    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
>    --- End of inner exception stack trace ---
>    at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, Utf8JsonReader& reader, Exception ex)
>    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
>    at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
>    at System.Text.Json.JsonSerializer.ContinueDeserialize[TValue](ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack, JsonConverter converter, JsonSerializerOptions options)
>    at System.Text.Json.JsonSerializer.ReadAllAsync[TValue](Stream utf8Json, JsonTypeInfo jsonTypeInfo, CancellationToken cancellationToken)
>    at Tvdb.Sdk.SearchClient.ReadObjectResponseAsync[T](HttpResponseMessage response, IReadOnlyDictionary`2 headers, CancellationToken cancellationToken)
>    --- End of inner exception stack trace ---
>    at Tvdb.Sdk.SearchClient.ReadObjectResponseAsync[T](HttpResponseMessage response, IReadOnlyDictionary`2 headers, CancellationToken cancellationToken)
>    at Tvdb.Sdk.SearchClient.GetSearchResultsByRemoteIdAsync(String remoteId, CancellationToken cancellationToken)
>    at Jellyfin.Plugin.Tvdb.TvdbClientManager.GetSeriesByRemoteIdAsync(String remoteId, String language, CancellationToken cancellationToken)
> ...
> ```
> 
> Post it as separate issue here: crobibero/tvdb-sdk-csharp?

This serialization problem is caused by the MovieBaseRecord and possibly MovieExtendedRecord schemas. Should little effect on the plugin since its scope is limited to Series only.

File: Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs (hunk 2 above)
Comment (BobSilent) [reply Jellyfin.Plugin.Tvdb/Providers/TvdbMissingEpisodeProvider.cs]: I will do so. it was used like that by intention. the templating has in most cases a value for structured logger, to have the information in an extra field. But the method name is not so valuable here to treat it like that. But anyway, at the end it is logged in the message.

**Review by crobibero — APPROVED** [review]

**Comment by crobibero** [comment]
Going to merge this and create an issue to track the deserialization issue

**Comment by BobSilent** [comment]
> > Shall I create a PR for this in jellyfin/jellyfin for the 10.8 branch?
> 
> Thank you for investigating the figuring out the logging issue! We aren't planning on making any more 10.8 releases, so a PR targeting master would be greatly appreciated!
> 
> > Post it as separate issue here: crobibero/tvdb-sdk-csharp?
> 
> That sdk is 99.97% auto-generated, any schema issues should be filed in the upstream repository (https://github.com/thetvdb/v4-api). The difficulty there lies with trying to figure out which model the issue is actually in
> 
> I think this PR can be merged without fixing the deserialization issues

Just for completeness I tested this on 10.9, here with the changes from https://github.com/jellyfin/jellyfin/pull/9078 this is fixed.
I have migrated the plugin to 10.9, i can share the code here.

**Comment by crobibero** [comment]
Feel free to open a PR here, I can just mark it as blocked until 10.9 is released

**Pull request merged**